ocamltest/filecompare.ml

```diff
@@ -53,22 +56,35 @@ type files = {
   output_filename : string;
 }
 
-let unixify_end_of_lines filename =
-  let commandline = "sed -i -e s/\\r$// " ^ filename in
-  let status = Run_command.run_commandline commandline in
-  (status, commandline)
+let read_file bytes_to_ignore filetype fn =
+  let ic = open_in_bin fn in
+  seek_in ic bytes_to_ignore;
+  let drop_cr s =
+    let l = String.length s in
+    if l > 0 && s.[l - 1] = '\r' then String.sub s 0 (l - 1)
+    else raise Exit
+  in
+  let rec loop acc =
+    match input_line ic with
```

Comment:
@damiendoligez wrote:
> I'm slightly nervous at the presence of both `bytes_to_ignore` and `input_line`. `bytes_to_ignore` exists because we want to compare binary executables, which are not text.
>
> `input_line` might fail on a 32-bit machine because of the string length limitation.
>
> More important, the automatic detection of line endings will be thrown off by binary files if they happen to contain some `\r\n` sequences as well as some `\n` characters on their own.

I don't *think* that the `input_line` logic can get confused here, because the file is opened using `open_in_bin` (so `\r` would just be returned in the string on all platforms).

The string limit on 32-bit is potentially a problem, I agree, because some of the binary files can end up being somewhat large.